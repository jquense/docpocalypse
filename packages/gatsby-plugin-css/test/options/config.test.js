const { webpack } = require('../helpers');

describe('Options', () => {
  test('Config - {Object}', () => {
    const config = {
      loader: {
        test: /\.css$/,
      },
    };

    return webpack('css/index.js', config).then((stats) => {
      const { source } = stats.toJson().modules[1];

      expect(source).toEqual(
        'module.exports = "a { color: rgba(255, 0, 0, 1.0) }\\n"',
      );

      expect(source).toMatchSnapshot();
    });
  });

  test('Config - Path - {String}', () => {
    const config = {
      loader: {
        test: /\.css$/,
        options: {
          config: { path: 'test/fixtures/config/postcss.config.js' },
        },
      },
    };

    return webpack('css/index.js', config).then((stats) => {
      const { source } = stats.toJson().modules[1];

      expect(source).toEqual('module.exports = "a { color: black }\\n"');
      expect(source).toMatchSnapshot();
    });
  });

  test('Config - Context - {Object}', () => {
    const config = {
      loader: {
        test: /\.css$/,
        options: {
          config: {
            path: 'test/fixtures/config/postcss.config.js',
            ctx: { plugin: true },
          },
        },
      },
    };

    return webpack('css/index.js', config).then((stats) => {
      const { source } = stats.toJson().modules[1];

      expect(source).toEqual(
        'module.exports = "a { color: rgba(255, 0, 0, 1.0) }\\n"',
      );

      expect(source).toMatchSnapshot();
    });
  });

  test('Config - Context - {Object} - with ident', () => {
    const config = {
      loader: {
        test: /\.css$/,
        options: {
          ident: 'postcss',
          config: {
            path: 'test/fixtures/config/postcss.config.js',
            ctx: { plugin: true },
          },
        },
      },
    };

    return webpack('css/index.js', config).then((stats) => {
      const { source } = stats.toJson().modules[1];

      expect(source).toEqual(
        'module.exports = "a { color: rgba(255, 0, 0, 1.0) }\\n"',
      );

      expect(source).toMatchSnapshot();
    });
  });

  test('Config – Context – Loader {Object}', () => {
    const config = {
      loader: {
        test: /\.css$/,
        options: {
          config: {
            path: 'test/fixtures/config/context/postcss.config.js',
          },
        },
      },
    };

    return webpack('css/index.js', config).then((stats) => {
      const { assets } = stats.compilation;

      const asset = 'asset.txt';

      expect(asset in assets).toBeTruthy();
      expect(assets[asset].source()).toBe('123');
    });
  });
});
