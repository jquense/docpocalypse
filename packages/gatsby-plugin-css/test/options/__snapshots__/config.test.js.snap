// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options Config - {Object} 1`] = `"module.exports = \\"a { color: rgba(255, 0, 0, 1.0) }\\\\n\\""`;

exports[`Options Config - Context - {Object} - with ident 1`] = `"module.exports = \\"a { color: rgba(255, 0, 0, 1.0) }\\\\n\\""`;

exports[`Options Config - Context - {Object} 1`] = `"module.exports = \\"a { color: rgba(255, 0, 0, 1.0) }\\\\n\\""`;

exports[`Options Config - Path - {String} 1`] = `"module.exports = \\"a { color: black }\\\\n\\""`;
